filament/src/materials/dof.mat

```diff
@@ -49,17 +44,271 @@ fragment {
 
 #include "dofUtils.fs"
 
+
+#define RING_DENSITY    6.0
+#define RING_COUNT	    4.0
+#define SAMPLE_COUNT    ((RING_DENSITY * RING_COUNT * (RING_COUNT - 1.0) + 2.0) * 0.5)
+
+void dummy(){}
+
+layout(location = 1) out float outAlpha;
+
+
+float sampleWeight(float coc) {
+    // The contribution of sample is inversely proportional to *it's* area
+    // (the larger area, the fainter it is).
+    // In theory this factor should be 1 / pi * radius^2, however 1/pi is a constant, and
+    // because we divide by the total weight eventually, it can be ommited, in fact, any
+    // constant can be used. Here we use MAX_COC_RADIUS * MAX_COC_RADIUS instead of 1/pi,
+    // because this allows us to fit in mediump precision.
+    return (MAX_COC_RADIUS * MAX_COC_RADIUS) / (max(coc * coc, 0.5));
+}
+
+float intersection(float border, float absCoc, float mip) {
+    return saturate((absCoc - border) * pow(0.5, mip) + 0.5);
+}
+
+float rcp(float x) {
```

Comment:
Should we just move this to `common_math.fs`?

Reply:
I thought of it but the reason I didn't do it is that I'm unclear about what happens with precision qualifiers. e.g. should it specify `highp`? but if it doesn't and it's fed a `highp` value, will it loose its `highp`ness?

Reply:
I mean that's true of all of our math helpers. They all assume default precision right now, we can always make a rcpHighp if needed.